packages/functional-tests/tests/react-conversion/signup.spec.ts

```diff
@@ -177,3 +189,131 @@ test.describe('severity-1 #smoke', () => {
+    });
+
+    test('Visits the terms of service links save information upon return', async ({
+      page,
+      pages: { signupReact },
+    }) => {
+      skipCleanup = true;
+      await signupReact.goto();
+      await signupReact.fillOutEmailFirst(email);
+      await signupReact.fillOutSignupForm(PASSWORD, '21', false);
+      await signupReact.visitTermsOfServiceLink();
+      await page.waitForURL(/legal\/terms/);
+      await page.goBack();
+
+      // TBD: Currently not supported. Do we need to support this?
+      // expect(await signupReact.getEmail().inputValue()).toEqual(email);
+      // expect(await signupReact.getPassword().inputValue()).toEqual(PASSWORD);
+      // expect(await signupReact.getPasswordConfirm().inputValue()).toEqual(PASSWORD);
+      // expect(await signupReact.getAge().inputValue()).toEqual('21');
+    });
+
+    test('Checks that form prefill information is cleared after sign up -> sign out', async ({
+      page,
+      target,
+      pages: { signupReact, settings },
+    }) => {
+      await signupReact.goto();
+      await signupReact.fillOutEmailFirst(email);
+      await signupReact.fillOutSignupForm(PASSWORD);
+      const code = await target.email.waitForEmail(
+        email,
+        EmailType.verifyShortCode,
+        EmailHeader.shortCode
+      );
+      await signupReact.fillOutCodeForm(code);
+      await page.waitForURL(/settings/);
+      await settings.signOut();
+      await signupReact.goto();
+
+      // TBD: No pre fill support currently. Do we even need this?
+      expect(await signupReact.getEmail().inputValue()).toEqual('');
+      await signupReact.fillOutEmailFirst('new-' + email);
+      expect(await signupReact.getPassword().inputValue()).toEqual('');
+      expect(await signupReact.getPasswordConfirm().inputValue()).toEqual('');
+      expect(await signupReact.getAge().inputValue()).toEqual('');
+    });
+
+    // TODO: Not currently supported
+    test('signup via product page and redirect after confirm', async ({
+      page,
+      target,
+      pages: { signupReact, relier, subscribe, login },
+    }) => {
+      // Make sure user is logged out
+      await login.clearCache();
+
+      // Go an RP's subscription page
+      await relier.goto();
+      await relier.clickSubscribe6Month();
+
+      // Click the sign in link
+      await subscribe.visitSignIn();
+
+      // Preserve search params but add in react experiment parameters
+      const searchParams = new URL(page.url()).searchParams;
+      await signupReact.goto('/', searchParams);
+      await signupReact.fillOutEmailFirst(email);
+      await signupReact.fillOutSignupForm(PASSWORD);
+      const code = await target.email.waitForEmail(
+        email,
+        EmailType.verifyShortCode,
+        EmailHeader.shortCode
+      );
+
+      await signupReact.fillOutCodeForm(code);
+      // TBD: The redirect back to products isn't working. Do we need a follow up for this?
```

Comment:
I think this should have been taken care of in #16078? Hmm.

Reply:
Last I tested it (last Friday) it was not working... Should we file another followup?

Reply:
https://mozilla-hub.atlassian.net/browse/FXA-8795